### PR TITLE
DAOS-1429: Add security cred to pool handle on targets

### PR DESCRIPTION
**Pull request by dpquigl**
In order to perform further security checks the security credential needs to be
added to the pool handle on the server side so it can be retrieved later.

Signed-off-by: David Quigley <david.quigley@intel.com>

Requesting exception as only test failures were due to a post provisioning bug in pipeline-lib https://github.com/daos-stack/pipeline-lib/commit/62633799c9795169a5806390ef443bfbcb238d65

### CODE REVIEW

File: src/pool/srv_pool.c [hunk 1]
```diff
@@ -1692,7 +1692,7 @@ ds_pool_create_handler(crt_rpc_t *rpc)
 
 static int
 pool_connect_bcast(crt_context_t ctx, struct pool_svc *svc,
-		   const uuid_t pool_hdl, uint64_t capas,
+		   const uuid_t pool_hdl, uint64_t capas, d_iov_t creds,
```
Comment (kjacque) [inline src/pool/srv_pool.c]: Doesn't look like we're actually using it yet in the code below. This function is where tci_cred should be populated, right?

File: src/pool/srv_layout.c [hunk 2]
```diff
@@ -49,6 +46,9 @@ RDB_STRING_KEY(ds_pool_prop_, nhandles);
 /** pool handle KVS */
 RDB_STRING_KEY(ds_pool_prop_, handles);
 
+/** Security credential handle KVS */
+RDB_STRING_KEY(ds_pool_sec_, handles);
+
```
Comment (kjacque) [inline src/pool/srv_layout.c]: Is this being used somewhere? Or just an artifact of an earlier implementation?

File: src/pool/srv.c [hunk 3]
```diff
@@ -77,7 +77,6 @@ fini(void)
 {
 	ds_pool_rsvc_class_unregister();
 	ds_pool_iv_fini();
-	ds_pool_hdl_hash_fini();
```
[the remark this answers is not in the document]
Comment (dpquigl) [reply src/pool/srv.c]: Good catch. I meant to remove something else instead.

File: src/pool/srv_layout.c (hunk 2 above)
Comment (dpquigl) [reply src/pool/srv_layout.c]: Good catch. Its an artifact of an earlier implementation.

File: src/pool/srv_pool.c (hunk 1 above)
Comment (dpquigl) [reply src/pool/srv_pool.c]: That is correct. THat must have gotten lost in one of my rebases.

File: src/pool/srv_target.c [hunk 4]
```diff
@@ -774,8 +775,15 @@ ds_pool_tgt_connect_handler(crt_rpc_t *rpc)
 	hdl->sph_capas = in->tci_capas;
 	hdl->sph_pool = pool;
 
+	rc = daos_iov_copy(&in->tci_cred, &hdl->sph_cred);
```
Comment (kjacque) [inline src/pool/srv_target.c]: Just realized, the first param of daos_iov_copy is supposed to be the destination. Everything else looks okay to me.

File: src/pool/srv_target.c [hunk 5]
```diff
@@ -774,8 +775,15 @@ ds_pool_tgt_connect_handler(crt_rpc_t *rpc)
 	hdl->sph_capas = in->tci_capas;
 	hdl->sph_pool = pool;
 
+	rc = daos_iov_copy(&hdl->sph_cred, &in->tci_cred);
+	if (rc != 0) {
+		ds_pool_put(pool);
+		D_GOTO(out, rc);
```
Comment (liw) [inline src/pool/srv_target.c]: D_FREE(hdl)?

File: src/pool/srv_target.c [hunk 6]
```diff
@@ -774,8 +775,15 @@ ds_pool_tgt_connect_handler(crt_rpc_t *rpc)
...
+	}
+
 	rc = pool_hdl_add(hdl);
 	if (rc != 0) {
+		daos_iov_free(&hdl->sph_cred);
 		ds_pool_put(pool);
 		D_GOTO(out, rc);
```
Comment (liw) [inline src/pool/srv_target.c]: Not a fault of this PR, but D_FREE(hdl) here?

Comment (dpquigl) [reply src/pool/srv_target.c]: I will make this fix as well and push a new update.

File: src/pool/srv_target.c (hunk 5 above)
Comment (dpquigl) [reply src/pool/srv_target.c]: Good catch. Ill add this and push a new update.

**Pull request merged**